src/fsharp/service/ServiceDeclarationLists.fs

```diff
@@ -650,22 +652,24 @@ type FSharpDeclarationListInfo(declarations: FSharpDeclarationListItem[], isForT
                             | Some _ -> displayName
                             | None -> Lexhelp.Keywords.QuoteIdentifierIfNeeded displayName
 
-                    let isAttribute = SymbolHelpers.IsAttribute infoReader item.Item
-                    
                     let cutAttributeSuffix (name: string) =
-                        if isAttributeApplicationContext && isAttribute && name <> "Attribute" && name.EndsWith "Attribute" then
```

Comment:
Wild that this call was a bottleneck. Good find.

Reply:
Accessibility rules by the signature file, those modifiers are left from the time when there is no one.

Reply:
Cutattributesuffix is called at least twice. How about using lazy expression here to make sure isAttribute is calculated at most once?

Reply:
@forki, it is called twice but on potentially different values (see how name and nameInCode are initialized), although it is right that we could optimize some of this because we never want to cut attribute suffix for those cases:

* operator
* back ticked name

I think it should apply only to `name` but not nameInCode which is different only for the cases mentioned above.

Reply:
I'm talking about the `SymbolHelpers.IsAttribute infoReader item.Item` part. That one was the bottleneck and stays stable over both calls. That's why it was factored out to the top. I propose we factor it back to the top but make it lazy.